test/integration/shoots/applications/metrics.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests that runtime metrics are available
+
+	AfterSuite
+		- Cleanup Workload in Shoot
+
+	Test: Create arbitrary deployment
+	Expected Output
+		- Metrics are available
+ **/
+
+package applications
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/test/framework"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/test/framework/resources/templates"
+	"github.com/onsi/ginkgo"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsv1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	metricsTestTimeout = 10 * time.Minute
+)
+
+var _ = ginkgo.Describe("Shoot application metrics testing", func() {
+
+	f := framework.NewShootFramework(&framework.ShootConfig{
+		CreateTestNamespace: true,
+	})
+
+	var (
+		name = "metrics-test"
+	)
+
+	f.Beta().CIt("should read runtime metrics", func(ctx context.Context) {
+		templateFilepath := filepath.Join(f.TemplatesDir, templates.SimpleLoadDeploymentName)
+		testTmpl, err := template.ParseFiles(templateFilepath)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Deploy the metrics test deployment")
+		var writer bytes.Buffer
+		err = testTmpl.Execute(&writer, map[string]string{
+			"name":      name,
+			"namespace": f.Namespace,
+		})
+		framework.ExpectNoError(err)
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = f.ShootClient.Applier().ApplyManifest(ctx, manifestReader, kubernetes.DefaultMergeFuncs)
+		framework.ExpectNoError(err)
```

Comment:
you could use this helper function to deploy and render the template https://github.com/gardener/gardener/blob/master/test/framework/template.go#L28 
Example: https://github.com/gardener/gardener/blob/master/test/framework/applications/guestbooktest.go#L169

```suggestion
		templateParams := map[string]string{
			"name":      name,
			"namespace": f.Namespace,
		}
		err = f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.SimpleLoadDeploymentName, templateParams )
		framework.ExpectNoError(err)
```

Reply:
Thank @schrodit, great advice. I changed the implementation for the [network test](https://github.com/gardener/gardener/blob/3576ef7931632d4a0f66ee46f942aaa3dee9bdbd/test/integration/shoots/applications/networking.go) as well. PTAL.